tensorflow/python/training/mixed_precision/experimental/loss_scale.py

```diff
@@ -0,0 +1,363 @@
+      graph = ops.get_default_graph()
+      graph_key = graph._graph_key # pylint: disable=protected-access
+
+    key = (name, graph_key)
+    if self._weights.get(key, None) is not None:
+      raise RuntimeError('Duplicate variables detected. {}'.format(key))
+    self._weights[key] = variable
+    self._handle_deferred_dependencies(name=name, trackable=variable)
+    return variable
+
+  @property
+  def _checkpoint_dependencies(self):
+    """From Trackable. Gather graph-specific weights to save."""
+    if context.executing_eagerly():
+      graph_key = None
+    else:
+      graph = ops.get_default_graph()
+      graph_key = graph._graph_key # pylint: disable=protected-access
+    weights = [trackable.TrackableReference(name=name, ref=v)
+               for (name, g), v in sorted(
+                   self._weights.items(), key=lambda i: i[0][0])
+               if g == graph_key]
+    return super(LossScale, self)._checkpoint_dependencies + weights
+
+  def _lookup_dependency(self, name):
+    """From Trackable. Find a weight in the current graph."""
+    unconditional = super(LossScale, self)._lookup_dependency(name)
+    if unconditional is not None:
+      return unconditional
+    if context.executing_eagerly():
+      graph_key = None
+    else:
+      graph = ops.get_default_graph()
+      graph_key = graph._graph_key # pylint: disable=protected-access
+    return self._weights.get((name, graph_key), None)
+
+@tf_export(v1=['train.mixed_precision.experimental.FixedLossScale'])
+class FixedLossScale(LossScale):
+  """Loss scale class with a fixed value.
+
+  The loss scale is not updated for the lifetime of the class.
+  """
+
+  def __init__(self, loss_scale_value):
+    """Creates the fixed loss scale.
+
+    Args:
+      loss_scale: A Python float. Its ideal value varies depending on models to
+        run. Choosing a too small loss_scale might affect model quality; a too
+        big loss_scale might cause inf or nan. There is no single right
+        loss_scale to apply. There is no harm choosing a relatively big number
+        as long as no nan or inf is encountered in training.
+
+    Raises:
+      ValueError: If loss_scale is less than 1.
+    """
+    super(FixedLossScale, self).__init__()
+    if not isinstance(loss_scale_value, six.integer_types + (float,)):
+      raise ValueError('loss_scale must be a Python int or float.')
+    if loss_scale_value < 1:
+      raise ValueError('loss scale must be at least 1.')
+    self._tensor_loss_scale = ops.convert_to_tensor(loss_scale_value,
+                                                    dtype=dtypes.float32)
+
+  def __call__(self):
+    return self._tensor_loss_scale
+
+  def update(self, grads):
+    del grads
+    return control_flow_ops.no_op(), True
+
+
+def _is_all_finite(grads):
+  """Returns a scalar boolean tensor indicating if all gradients are finite."""
+  is_finite_per_grad = [math_ops.reduce_all(math_ops.is_finite(g))
+                        for g in grads]
```

Comment:
If g is none in this comprehension then this results in an error.

Reply:
Thanks for pointing this out; we have a new PR proposing a patch here: #27898